### ISSUE TITLE
Bad highlighting with high unicode characters

### ISSUE THREAD

**Issue by yannl35133** [ocamllabs/vscode-ocaml-platform#708]
When high unicode characters are used, it messes up with highlighting and errors placed in text (these are capital pis):
```ocaml
let _ =
  let _ = "ΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠΠ" in -. 0                        
```
![Screenshot_20210915_165450](https://user-images.githubusercontent.com/40719961/133457718-941d7cb8-54b8-4789-b72c-ec2753deb129.png)

**Comment by tmattio** [comment]
@rgrinberg This is something that should be addressed in LSP, right? Any idea what could cause this?

**Comment by rgrinberg** [comment]
Yeah, probably. There's not much that can be done in this repo regardless.

**Comment by smorimoto** [comment]
Confirmed with ocamllsp 1.27.0: the server negotiates UTF-16 but reports Merlin's UTF-8 byte offsets unchanged, so diagnostics after non-ASCII characters are misplaced. This is an upstream ocaml-lsp issue, tracked in ocaml/ocaml-lsp#304, and cannot be fixed in this extension. Closing in favour of the upstream issue.
